cli/verifier/servicemesh_egress.go

```diff
@@ -0,0 +1,61 @@
+// Copyright 2025 Edgeless Systems GmbH
+// SPDX-License-Identifier: BUSL-1.1
+
+package verifier
+
+import (
+	"errors"
+
+	"github.com/edgelesssys/contrast/internal/kuberesource"
+)
+
+// ServiceMeshEgressNotEmpty verifies that the `contrast.edgeless.systems/servicemesh-egress` annotation
+// isn't empty if it exists.
+type ServiceMeshEgressNotEmpty struct{}
+
+// Verify verifies that the `contrast.edgeless.systems/servicemesh-egress` annotation
+// isn't empty if it exists.
+func (v *ServiceMeshEgressNotEmpty) Verify(toVerify any) error {
+	var findings error
+
+	resources, err := kuberesource.ResourcesToUnstructured([]any{toVerify})
+	if err != nil {
+		return err
+	}
+	for _, r := range resources {
+		findings = errors.Join(findings, checkUnstructured(r.UnstructuredContent()))
+	}
+
+	return findings
+}
+
+// checkUnstructured recursively checks if the "contrast.edgeless.systems/servicemesh-egress" annotation has content.
+func checkUnstructured(r map[string]any) error {
```

Comment:
The name should be more descriptive because we're in the shared `verifiers` package.